### PR TITLE
etcdctl:  prints name of key that changed when watching recursively

### PR DESCRIPTION
**Pull request by mgoelzer**
Previously 'etcdctl watch /DIR --recursive' would print only the new value
of any changed key, but not show which key actually changed.  Now it formats
its output as 'name_of_changed_key:  new_value'.  Only affects --recursive
watch; regular watches are not affected.  This matches the behavior of
'grep -r'.

Fixes issue #1520 in the manner suggested by philips

### CODE REVIEW

**Comment by kelseyhightower** [comment]
@mgoelzer Can you clean up the commit message a bit? I would like to suggest:

```
etcdctl: prints name of key that changed when watching recursively

Previously recursive watches would print only the new value
of any changed key, but not show which key actually changed.

For example, running the following watch command

    etcdctl watch /dir --recursive

Update a key under dir:

    etcdctl set /dir/key1 value1 

The above recursive watch would return the following:

    value1

Fix this issue by printing the key along with the value for recursive watches.

    /dir/key1:  value1

This change does not effect non-recursive watches.  This matches the behavior of 'grep -r'.

Fixes issue #1520
```

File: etcdctl/command/handle.go [hunk 1]
```diff
@@ -130,15 +130,27 @@ func printAll(resp *etcd.Response, format string) {
...
 
 // printKeyOnly only supports to print key correctly.
-func printKeyOnly(resp *etcd.Response, format string) {
+func printKeyOnly(resp *etcd.Response, format string, prependKey bool) {
 	// Format the result.
 	switch format {
 	case "simple":
-		fmt.Println(resp.Node.Value)
+		if prependKey {
+			fmt.Println(fmt.Sprintf("%s: %s", resp.Node.Key, resp.Node.Value))
```
Comment (kelseyhightower) [inline etcdctl/command/handle.go]: I think we can drop the extra call to `fmt.Sprintf` and just use `fmt.Printf` instead of `fmt.Println`.

**Comment by mgoelzer** [comment]
Kelsey, thanks for the feedback.  I amended the commit message as you suggested and then added a second commit that removes the extraneous fmt.Sprintf call you pointed out.

Let me know if you'd rather that I open a new PR so that everything is in a single commit.  I wasn't sure which way would be preferred.

**Comment by xiang90** [comment]
@mgoelzer I am closing this since etcdctl already starts to print out the key and action when `--recursive` is set.

Example
```
[set] /foo/bar
foo
```

**Pull request closed without merging**